mirgecom/initializers.py

```diff
@@ -840,3 +842,136 @@ def __call__(self, x_vec, eos, *, t=0.0):
 
         return join_conserved(dim=self._dim, mass=mass, energy=energy,
                               momentum=mom, species_mass=specmass)
+
+
+class PlanarDiscontinuity:
+    r"""Solution initializer for flow with a discontinuity.
+
+    This initializer creates a physics-consistent flow solution
+    given an initial thermal state (pressure, temperature) and an EOS.
+
+    The solution varies across a planar interface defined by a tanh function
+    located at disc_location for pressure, temperature, velocity, and mass fraction
+
+    .. automethod:: __init__
+    .. automethod:: __call__
+    """
+
+    def __init__(
+            self, *, dim=3, normal_dir=0, disc_location=0, nspecies=0,
+            temperature_left, temperature_right,
+            pressure_left, pressure_right,
+            velocity_left=None, velocity_right=None,
+            species_mass_left=None, species_mass_right=None,
+            convective_velocity=None, sigma=0.5
+    ):
+        r"""Initialize mixture parameters.
+
+        Parameters
+        ----------
+        dim: int
+            specifies the number of dimensions for the solution
+        normal_dir: int
+            specifies the direction (plane) the discontinuity is applied in
+        disc_location: float or Function[float]
+           location of discontinuity (in time)
+        nspecies: int
+            specifies the number of mixture species
+        pressure_left: float
+            pressure to the left of the discontinuity
+        temperature_left: float
+            temperature to the left of the discontinuity
+        velocity_left: numpy.ndarray
+            velocity (vector) to the left of the discontinuity
+        species_mass_left: numpy.ndarray
+            species mass fractions to the left of the discontinuity
+        pressure_right: float
+            pressure to the right of the discontinuity
+        temperature_right: float
+            temperaure to the right of the discontinuity
+        velocity_right: numpy.ndarray
+            velocity (vector) to the right of the discontinuity
+        species_mass_right: numpy.ndarray
+            species mass fractions to the right of the discontinuity
+        sigma: float
+           sharpness parameter
```

Comment:
Maybe specify the `tanh` blend curve so that the semantics for this can be defined?